File: packages/types/src/alibaba.ts
/**
 * Alibaba / sourcing types — snake_case matching Python scraper Pydantic models.
 * Single source of truth: apps/scraper/app/models/alibaba.py
 */

// ---------------------------------------------------------------------------
// Core types — map 1:1 to Python AlibabaSupplierBasic / AlibabaProductResult
// ---------------------------------------------------------------------------

export interface PriceRange {
    min?: number | null
    max?: number | null
    currency: string
}

export interface AlibabaSupplierBasic {
    id: string
    name: string
    country?: string | null
    verified: boolean
    years_on_platform?: number | null
    response_rate?: number | null
    url?: string | null
    logo_url?: string | null
}

/** A single product result from Alibaba — maps to Python AlibabaProductResult */
export interface AlibabaProductResult {
    id: string
    title: string
    supplier: AlibabaSupplierBasic
    min_order_quantity?: number | null
    price_range?: PriceRange | null
    image_url?: string | null
    url?: string | null
}

/** Maps to Python AlibabaSearchResponse */
export interface AlibabaSearchResponse {
    query: string
    page: number
    total: number
    results: AlibabaProductResult[]
}

// ---------------------------------------------------------------------------
// Legacy aliases — keep old names compiling during transition
// ---------------------------------------------------------------------------
/** @deprecated use AlibabaProductResult */
export type AlibabaProduct = AlibabaProductResult
/** @deprecated use AlibabaSearchResponse */
export type AlibabaSearchResult = AlibabaSearchResponse
/** @deprecated use AlibabaSupplierBasic */
export type AlibabaSupplier = AlibabaSupplierBasic & {
    supplierId: string
    supplierName: string
}

File: apps/scraper/app/models/alibaba.py
from __future__ import annotations
from pydantic import BaseModel


class PriceRange(BaseModel):
    min: float | None = None
    max: float | None = None
    currency: str = "USD"


class AlibabaSupplierBasic(BaseModel):
    id: str
    name: str
    country: str | None = None
    verified: bool = False
    years_on_platform: int | None = None
    response_rate: float | None = None
    url: str | None = None
    logo_url: str | None = None


class AlibabaProductResult(BaseModel):
    id: str
    title: str
    supplier: AlibabaSupplierBasic
    min_order_quantity: int | None = None
    price_range: PriceRange | None = None
    image_url: str | None = None
    url: str | None = None


class AlibabaSearchResponse(BaseModel):
    query: str
    page: int
    total: int
    results: list[AlibabaProductResult]
